hash/spectrum_cass.xml

```diff
@@ -1390,57 +1432,95 @@ license:CC0-1.0
 		<description>Arkanoid (set 1)</description>
 		<year>1987</year>
 		<publisher>Imagine Software</publisher>
+		<sharedfeat name="compatibility" value="48K" />
```

Comment:
Agreed about 48k. 16k machines shouldn't include this list. 128k extension is 48k compatible most of the cases anyway.
Hence 128+ only can have similar label, but not 48

Reply:
I understand all these considerations.
_It isn’t clear what you mean – are you saying it requires at least 48K RAM, or that it only works with 48K models and not 128K models?_ It means that the software was designed to work with this memory size. For instance, I tested some games, I don't remember which ones, that worked with 48K but didn't work with 128K.
But to avoid any confusion, I will move these info to the `usage` .

Reply:
> 16K versus 48K is a RAM size option, not a separate machine configuration, so it can’t have different compatibility flags on the software list.

Since the 16K and 48K models were actually sold as separate machines just like Amstrad's `CPC 664` and `CPC 6128`, perhaps MAME is just missing a `ZX Spectrum (16K)` which could make use of these compatibility flags?

For the record, Spectrum Computing already has an equivalent for these flags for every listed game (in each entry's "Machine" field) which does include 16K computers.

Reply:
> Since the 16K and 48K models were actually sold as separate machines just like Amstrad's `CPC 664` and `CPC 6128`, perhaps MAME is just missing a `ZX Spectrum (16K)` which could make use of these compatibility flags?

They’re essentially the same thing, which is why it’s a `-ramsize` option and not a separate machine.

> For the record, Spectrum Computing already has an equivalent for these flags for every listed game (in each entry's "Machine" field) which does include 16K computers.

Yeah, but MAME has a whole pile of additional classes of ZX Spectrum derivatives, and @holub keeps expanding the collection.  The of things to test compatibility with, and the size of the compatibility flags, would blow out very quickly.

Reply:
That `-ramsize` option needs to be present in the UI or 99% of the MAME userbase won't know it exists.

Reply:
That `-ramsize` option needs to be in the UI or 99% of the MAME userbase won't know it exists.